app/src/main/java/io/appium/settings/recorder/RecorderService.java

```diff
@@ -0,0 +1,192 @@
+            return START_NOT_STICKY;
+        }
+        final String action = intent.getAction();
+        if (action == null) {
+            Log.e(TAG, "onStartCommand: Unable to retrieve recording intent:action");
+            return START_NOT_STICKY;
+        }
+
+        int result = START_STICKY;
+        if (ACTION_RECORDING_START.equals(action)) {
+            showNotification(); // TODO is this really necessary
+
+            MediaProjectionManager mMediaProjectionManager =
+                    (MediaProjectionManager) getSystemService(Context.MEDIA_PROJECTION_SERVICE);
+
+            if (mMediaProjectionManager != null) {
+                startScreenRecord(mMediaProjectionManager, intent);
+            } else {
+                Log.e(TAG, "onStartCommand: " +
+                        "Unable to retrieve MediaProjectionManager instance");
+                result = START_NOT_STICKY;
+            }
+        } else if (ACTION_RECORDING_STOP.equals(action)) {
+            Log.v(TAG, "onStartCommand: Received recording stop intent, stopping recording");
+            stopScreenRecord();
+            result = START_NOT_STICKY;
+        } else {
+            Log.v(TAG, "onStartCommand: Received unknown recording intent with action: "
+                    + action);
+            result = START_NOT_STICKY;
+        }
+
+        return result;
+    }
+
+    /**
+     * start screen recording
+     */
+    @RequiresApi(api = Build.VERSION_CODES.Q)
+    private void startScreenRecord(MediaProjectionManager mediaProjectionManager,
+                                   final Intent intent) {
+        if (recorderThread != null) {
+            if (recorderThread.isRecordingRunning()) {
+                Log.v(TAG, "Recording is already continuing, exiting");
+                return;
+            } else {
+                Log.w(TAG, "Recording is stopped, " +
+                        "but recording instance is still alive, starting recording");
+                recorderThread = null;
+            }
+        }
+
+        int resultCode = intent.getIntExtra(ACTION_RECORDING_RESULT_CODE, 0);
+        // get MediaProjection
+        final MediaProjection projection = mediaProjectionManager.getMediaProjection(resultCode,
+                intent);
+        if (projection == null) {
+            Log.e(TAG, "Recording is stopped, Unable to retrieve MediaProjection instance");
+            return;
+        }
+
+        String outputFilePath = intent.getStringExtra(ACTION_RECORDING_FILENAME);
+        if (outputFilePath == null) {
+            Log.e(TAG, "Recording is stopped, Unable to retrieve outputFilePath instance");
+            return;
+        }
+
+        DisplayMetrics metrics = getResources().getDisplayMetrics();
+        int rawWidth = metrics.widthPixels;
+        int rawHeight = metrics.heightPixels;
+
+        int recordingRotation = intent.getIntExtra(ACTION_RECORDING_ROTATION, -1);
+
+        if (recordingRotation == -1) {
```

Comment:
can we get the orientation value from resources: https://github.com/appium/appium-uiautomator2-server/blob/5d49661b6198a52c59e869005a66e9a0f589c432/app/src/main/java/io/appium/uiautomator2/model/ScreenOrientation.java#L51 ?

Reply:
we can probably use but in my opinion our current method handles orientation more accurate and with more coverage (with exact 4 modes, 0, 90, 180 and 270) ref: https://stackoverflow.com/a/3663693